mender-deb-package

```diff
@@ -196,44 +200,76 @@ build_packages() {
     dh_make --file /orig/"${DEB_PACKAGE}_${DEB_VERSION_NO_REV}".orig.tar.gz --packagename "${DEB_PACKAGE}_${DEB_VERSION_NO_REV}" -s -y || true
   fi
 
+  dpkg_build_ignore_flag="--diff-ignore=(.git|vendor)"
+
+
+  skip_patches_flag="--source-option=--skip-patches"
+  if [ "$OS_DISTRO" = "debian" -a "$OS_CODENAME" = "buster" ]; then
+    echo "deb http://deb.debian.org/debian buster-backports main" > /etc/apt/sources.list.d/buster-backports_deps.list
+    cat <<EOF> /etc/apt/preferences.d/prefer_backports.pref
+Package: cmake*
+Pin: release a=buster-backports
+Pin-Priority: 999
+EOF
+    apt update
+    # We want the patches applied for Debian Buster
+    rm -f debian/source/local-options
+    skip_patches_flag=
+  else
+    # Skip applying the patches
+    rm -f debian/patches/series
+  fi
+
+  # Install the builder architecture independent dependencies from the debian/control file
+  mk-build-deps --install --build-indep --tool='apt-get -o Debug::pkgProblemResolver=yes --no-install-recommends --yes' debian/control
+
+  # HOST - the machine arch we are building _for_
+  HOST_ARCH=${ARCH}
+
+  dpkg --add-architecture ${ARCH}
+  apt update
+
+  mk-build-deps --install \
+                --build-dep \
+                --tool='apt-get -o Debug::pkgProblemResolver=yes --no-install-recommends --yes' \
+                --host-arch ${HOST_ARCH} \
+                debian/control
+
   case "$ARCH" in
     amd64)
       # Native build (amd64)
       dpkg-buildpackage \
         ${sign_flags} \
+        "${dpkg_build_ignore_flag}" \
         --build=$DEB_BUILD_TYPE
       ;;
 
     armhf)
       if [ "$OS_DISTRO" = "debian" ]; then
         # Custom toolchain to support ARMv6
+        PATH="${ARMV6_TOOLCHAIN_ROOT}/bin:$PATH" \
         CROSS_COMPILE="arm-buildroot-linux-gnueabihf" \
                     CC="$CROSS_COMPILE-gcc" \
-                    PATH="$PATH:${ARMV6_TOOLCHAIN_ROOT}/bin" \
-                    LD_LIBRARY_PATH="$LD_LIBRARY_PATH:${ARMV6_TOOLCHAIN_ROOT}/lib:${ARMV6_TOOLCHAIN_ROOT}/usr/lib" \
-                    CGO_CFLAGS="-I${ARMV6_TOOLCHAIN_ROOT}/usr/include \
-                                -I${ARMV6_TOOLCHAIN_ROOT}/usr/include/arm-linux-gnueabihf \
-                                -pthread \
-                                -I${ARMV6_TOOLCHAIN_ROOT}/usr/include/glib-2.0 \
-                                -I${ARMV6_TOOLCHAIN_ROOT}/usr/lib/arm-linux-gnueabihf/glib-2.0/include/" \
-                    CGO_LDFLAGS="-L${ARMV6_TOOLCHAIN_ROOT}/usr/lib/arm-linux-gnueabihf \
-                                -lssl -lcrypto \
-                                -lz -lgmodule-2.0 -lffi -lmount -lselinux -lpcre2-8 -lblkid -luuid \
-                                -lglib-2.0 -lpcre -lresolv -lgio-2.0 -lgobject-2.0 -lgmodule-2.0" \
+                    CXX=${CROSS_COMPILE}-g++ \
+                    CXXFLAGS=" -Wno-psabi" \
```

Comment:
https://gcc.gnu.org/gcc-7/changes.html

Reply:
I would add this link as a comment in the script, somewhere.